src/Modules/SimplCommerce.Module.WishList/Controllers/WishListController.cs

```diff
@@ -0,0 +1,324 @@
+                await _wishListRepository.SaveChangesAsync();
+            }
+
+            var wishlistVm = await List(wishList, pageNumber, pageSize);
+
+            return View(wishlistVm);
+        }
+
+        [Route("wishlist/list/{id}/{pageNumber?}/{pageSize?}")]
+        public async Task<IActionResult> PublicList(string id, int? pageNumber, int? pageSize)
+        {
+            var user = await _workContext.GetCurrentUser();
+
+            var wishList = await _wishListRepository
+                .Query()
+                .SingleOrDefaultAsync(x => x.SharingCode == id);
+
+            if (wishList == null)
+            {
+                return new NotFoundResult();
+            }
+
+            var wishlistVm = await List(wishList, pageNumber, pageSize);
+
+            return View(wishlistVm);
+        }
+
+        private async Task<WishListVm> List(Models.WishList wishList, int? pageNumber, int? pageSize)
+        {
+            var itemsPerPage = pageSize.HasValue ? pageSize.Value : DefaultPageSize;
+            var currentPageNum = pageNumber.HasValue ? pageNumber.Value : 1;
+            var offset = (itemsPerPage * currentPageNum) - itemsPerPage;
+
+            var wishlistVm = new WishListVm()
+            {
+                Id = wishList.Id,
+                SharingCode = wishList.SharingCode
+            };
+
+            var wishListItemsQuery = _wishListItemRepository
+                .Query()
+                .Where(x => x.WishListId == wishList.Id)
+                .Select(x => new WishListItemVm()
+                {
+                    Id = x.Id,
+                    WishListId = x.WishListId,
+                    ProductId = x.Product.Id,
+                    ProductName = x.Product.Name,
+                    Description = x.Description,
+                    ProductImage = _mediaService.GetThumbnailUrl(x.Product.ThumbnailImage),
+                    Quantity = x.Quantity
+                });
+
+            wishlistVm.Items.Data = await wishListItemsQuery
+                .Skip(offset)
+                .Take(itemsPerPage)
+                .ToListAsync();
+
+            wishlistVm.Items.PageNumber = currentPageNum;
+            wishlistVm.Items.PageSize = itemsPerPage;
+            wishlistVm.Items.TotalItems = await wishListItemsQuery.CountAsync();
+
+            return wishlistVm;
+        }
+
+        [Authorize]
+        [HttpPost]
+        public async Task<IActionResult> AddItem([FromBody] AddToWishList model)
+        {
+            if (ModelState.IsValid)
+            {
+                var user = await _workContext.GetCurrentUser();
+                var resultModel = new AddToWishListResult();
+
+                var product = await _productRepository
+                    .Query()
+                    .Include(x => x.ThumbnailImage)
+                    .SingleOrDefaultAsync(x => x.Id == model.ProductId);
+
+                if (product == null)
+                {
+                    return new NotFoundResult();
```

Comment:
should be return NotFound(); for consistency and similar for other places

Reply:
@thiennn 

Do you mean to replace all places in the controller where NotFoundResult() is used with NotFound(), or only in the two places you actually pointed out?

Also, when do you prefer to use NotFoundResult() versus NotFound()? I saw a lot of places in the existing code that are using NotFoundResult().

Thanks.

Reply:
All the places should use NotFound(); I will eventually update to NoFound() whenever I have a change.